kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,243 @@
+            }
+        });
+
+        buttonCloseFragment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        return view;
+    }
+
+    /**
+     * toggle button as selected or deselected
+     * @param select true or false
+     * @param feedbackRating the views to toggle
+     */
+    void toggleRatingButtonSelected(boolean select, FeedbackRatingGroup.FeedbackRating feedbackRating) {
+        //set drawable
+        feedbackRatingGroup.setDrawableForRating(select, feedbackRating);
+
+        //set visibility and size
+        if(select) {
+            feedbackRating.ratingButton.setScaleX(1f);
+            feedbackRating.ratingButton.setScaleY(1f);
+            feedbackRating.feedbackTextView.setVisibility(View.VISIBLE);
+        } else {
+
+            feedbackRating.ratingButton.setScaleX(0.8f);
+            feedbackRating.ratingButton.setScaleY(0.8f);
+            feedbackRating.feedbackTextView.setVisibility(View.GONE);
+        }
+    }
+
+    //when one of the rating buttons is clicked
+    @Override
+    public void onClick(View view) {
+        Integer buttonTag = (Integer) view.getTag();
+        setRatingLevel(buttonTag);
+
+        //show the feedback comment input edit text, if not already visible
+        if(editTextFeedbackComment.getVisibility()==View.GONE) {
+            editTextFeedbackComment.setVisibility(View.VISIBLE);
+        }
+
+        for(FeedbackRatingGroup.FeedbackRating feedbackRating : feedbackRatingGroup.feedbackRating) {
+            Integer iterationButtonTag = (Integer) feedbackRating.ratingButton.getTag();
+            toggleRatingButtonSelected(iterationButtonTag.intValue() == buttonTag.intValue(), feedbackRating);
+        }
+    }
+
+    /**
+     * class for the feedback rating input buttons and text views
+     */
+    public class FeedbackRatingGroup {
+        Drawable[] selectedDrawable;
+        Drawable[] unSelectedDrawable;
+        FeedbackRating[] feedbackRating;
+        int noOfRatingElements;
+
+        public FeedbackRatingGroup(int noOfRatingElements, Drawable[] selectedDrawable, Drawable[] unSelectedDrawable) {
+            this.noOfRatingElements = noOfRatingElements;
+            this.selectedDrawable = selectedDrawable;
+            this.unSelectedDrawable = unSelectedDrawable;
+            feedbackRating = new FeedbackRating[noOfRatingElements];
+        }
+
+        /**
+         * class for a single rating button, text view and their properties
+         */
+        class FeedbackRating {
+            Button ratingButton;
+            TextView feedbackTextView;
+            int ratingValue;
+        }
+
+        void setDrawableForRating(boolean select, FeedbackRating feedbackRating) {
+            if(select) {
+                feedbackRating.ratingButton.setBackground(selectedDrawable[feedbackRating.ratingValue - 1]);
```

Comment:
@shubhamtewari as the operation and the values on the basis of which some operation is happening both are in the same class so the whole logic should be moved to feedbackRating itself. we can have something like feedbackRating.selectDrawable() and that method internally can set the required background.